main.go

```diff
@@ -43,21 +44,36 @@ type CLI struct {
 
 func (cli *CLI) build(args ...string) {
 	api := api.New(cli.config, ctx)
-	build_workers := len(cli.config.Extensions)
+
+	var wg sync.WaitGroup
 	build_chan := make(chan build.Result)
 
+	errors := 0
 	for _, e := range cli.config.Extensions {
 		b := build.NewBuilder(e)
-		log.Printf("Building %s, id: %s", e.Type, e.UUID)
 
+		wg.Add(1)
 		go b.Build(ctx, func(result build.Result) {
 			build_chan <- result
+			log.Printf("EXT LEN %d", len(cli.config.Extensions))
+			if !result.Success {
+				errors++
+				log.Printf("[Build] Error: %s, Extension: %s", result.Error, result.UUID)
+			} else {
+				log.Printf("[Build] Success! Extension: %s", result.UUID)
+			}
 		})
 
-		go cli.monitor(build_workers, build_chan, "Build", api, e)
+		go cli.monitor(wg, build_chan, "Build", api, e)
 	}
 
-	<-build_chan
+	wg.Wait()
+
+	if errors > 0 {
```

Comment:
this is how I was checking for build errors before, waiting for all extensions to build, then checking if there were errors, but this approach doesn't work well with the goroutines.

I could do a similar check within the loop, but that also doesn't work well because we have 2 other goroutines in the loop.